src/main/java/org/opensearch/ad/ratelimit/EntityColdStartWorker.java

```diff
@@ -114,15 +120,36 @@ protected void executeRequest(EntityRequest coldStartRequest, ActionListener<Voi
             0
         );
 
-        ActionListener<Void> failureListener = ActionListener.delegateResponse(listener, (delegateListener, e) -> {
-            if (ExceptionUtil.isOverloaded(e)) {
-                LOG.error("OpenSearch is overloaded");
-                setCoolDownStart();
+        ActionListener<Void> coldStartListener = ActionListener.wrap(r -> {
+            nodeStateManager.getAnomalyDetector(detectorId, ActionListener.wrap(detectorOptional -> {
+                try {
+                    if (!detectorOptional.isPresent()) {
+                        LOG.error(new ParameterizedMessage("fail to get detector [{}]", detectorId));
+                        return;
+                    }
+                    AnomalyDetector detector = detectorOptional.get();
+                    EntityModel model = modelState.getModel();
+                    // load to cache if cold start succeeds
+                    if (model != null && model.getTrcf() != null) {
+                        cacheProvider.get().hostIfPossible(detector, modelState);
```

Comment:
previously did we not load to cache? Cant find where this logic was previously if it was

Reply:
Previously, we don't load to cache when cold start finishes. We would have to wait for the next interval to make it happen.